singularity-eos/eos/eos_noble_abel.hpp

```diff
@@ -0,0 +1,210 @@
+// © 2021-2023. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#ifndef _SINGULARITY_EOS_EOS_EOS_NOBLE_ABEL_HPP_
+#define _SINGULARITY_EOS_EOS_EOS_NOBLE_ABEL_HPP_
+
+// stdlib
+#include <cmath>
+#include <cstdio>
+#include <string>
+
+// Ports-of-call
+#include <ports-of-call/portability.hpp>
+
+// Base stuff
+#include <singularity-eos/base/constants.hpp>
+#include <singularity-eos/base/eos_error.hpp>
+#include <singularity-eos/base/robust_utils.hpp>
+#include <singularity-eos/eos/eos_base.hpp>
+
+namespace singularity {
+
+using namespace eos_base;
+
+class NobleAbel : public EosBase<NobleAbel> {
+ public:
+  NobleAbel() = default;
+  PORTABLE_INLINE_FUNCTION NobleAbel(Real gm1, Real Cv, Real bb, Real qq)
+      : _Cv(Cv), _gm1(gm1), _bb(bb), _qq(qq), _T0(ROOM_TEMPERATURE),
+        _P0(ATMOSPHERIC_PRESSURE), _qp(0.0),
+        _rho0(robust::ratio(_P0, gm1 * Cv * _T0 + bb * _P0)),
+        _vol0(robust::ratio(gm1 * Cv * _T0 + bb * _P0, _P0)), _sie0(Cv * _T0 + qq),
+        _bmod0(robust::ratio(_rho0 * Cv * _T0 * gm1 * (gm1 + 1.0),
+                             (1.0 - bb * _rho0) * (1.0 - bb * _rho0))),
+        _dpde0(robust::ratio(_rho0 * gm1, 1.0 - bb * _rho0)) {
+    checkParams();
+  }
+  PORTABLE_INLINE_FUNCTION NobleAbel(Real gm1, Real Cv, Real bb, Real qq, Real qp,
+                                     Real T0, Real P0)
+      : _Cv(Cv), _gm1(gm1), _bb(bb), _qq(qq), _T0(T0), _P0(P0), _qp(qp),
+        _rho0(robust::ratio(P0, gm1 * Cv * T0 + bb * P0)),
+        _vol0(robust::ratio(gm1 * Cv * T0 + bb * P0, P0)), _sie0(Cv * T0 + qq),
+        _bmod0(robust::ratio(_rho0 * Cv * T0 * gm1 * (gm1 + 1.0),
+                             (1.0 - bb * _rho0) * (1.0 - bb * _rho0))),
+        _dpde0(robust::ratio(_rho0 * gm1, 1.0 - bb * _rho0)) {
+    checkParams();
+  }
+  NobleAbel GetOnDevice() { return *this; }
+  PORTABLE_INLINE_FUNCTION Real TemperatureFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    return std::max(robust::SMALL(), (sie - _qq) / _Cv);
+  }
+  PORTABLE_INLINE_FUNCTION void checkParams() const {
+    PORTABLE_ALWAYS_REQUIRE(_Cv >= 0, "Heat capacity must be positive");
```

Comment:
```suggestion
    PORTABLE_ALWAYS_REQUIRE(_Cv > 0, "Heat capacity must be positive");
```
Since we divide by the heat capacity, let's ensure it's strictly positive